flax/linen/combinators.py

```diff
@@ -0,0 +1,39 @@
+"""Combinators of modules, such as a Sequential."""
+
+from typing import Callable, Sequence
+
+from flax.linen.module import Module
+
+class Sequential(Module):
+  """Applies a linear chain of Modules.
+
+  Meant to be used only for the simple case of fusing together callables where
+  the input of a particular module/op is the output of the previous one.
+
+  Modules will be applied in the order that they are passed in the constructor.
+
+  The apply() method of Sequential accepts any input and forwards it to the
+  first module it contains. It chains the output sequentially to the input of
+  the next module and returns the output of the final module.
+
+  Example usage::
+
+    class Foo(nn.Module):
+        feature_sizes: Sequence[int]
+
+        @nn.compact
+        def __call__(self, x):
+        return nn.Sequential([nn.Dense(layer_size, name=f'layers_{idx}')
```

Comment:
Can we make the example include a non-linearity like `relu`? That's going to be necessary for pretty much all use-cases of `nn.Sequential` and people will want to see how it's done.

Reply:
Ok, doing this in #1978 .